services/apps/alcs/src/portal/application-submission-draft/application-submission-draft.service.ts

```diff
@@ -159,6 +159,50 @@ export class ApplicationEditService {
     return submission;
   }
 
+  async publish(fileNumber: string, user: User) {
+    const draft = await this.getDraft(fileNumber);
+    if (!draft) {
+      throw new BaseServiceException(
+        `Failed to find Draft to Publish with File Number ${fileNumber}`,
+      );
+    }
+
+    const original = await this.applicationSubmissionRepository.findOne({
+      where: {
+        fileNumber,
+        isDraft: false,
+      },
+      relations: {
+        owners: true,
+        parcels: true,
+      },
+    });
+
+    if (!original) {
+      throw new BaseServiceException(
+        `Failed to find original submission to Publish with FileID ${fileNumber}`,
+      );
+    }
+
+    //Delete Original Submission
+    for (const owner of original.owners) {
+      await this.applicationOwnerService.delete(owner);
+    }
+    const parcelUuids = original.parcels.map((parcel) => parcel.uuid);
+    await this.applicationParcelService.deleteMany(parcelUuids);
+    await this.applicationSubmissionRepository.remove(original);
```

Comment:
I assume that the log table persists the removed submission well?

Reply:
yes! It will track the delete statement with the deleted row :)